src/Atlas.cc

```diff
@@ -321,7 +322,9 @@ void Atlas::PreSave()
     std::set<GeometricCamera*> spCams(mvpCameras.begin(), mvpCameras.end());
     for(Map* pMi : mvpBackupMaps)
     {
-        lock_guard<mutex> lock(pMi->mMutexMapUpdate);
+        // Lock everything that isn't the current map because that is already locked by the caller
+        mutex unusedMutex;
+        lock_guard<mutex> lock(pMi == mpCurrentMap ? unusedMutex : pMi->mMutexMapUpdate);
```

Comment:
It's a little hard to understand that when it's not the current map, we can lock mMutexMapUpdate while holding mMutexAtlas. But in general, the locking semantics of orbslam are not very clear, so this seems fine.

Reply:
I agree -- that line is pretty ugly. I would have liked to do this with an if statement like:
```
lock_guard<mutex> lock;
// Lock everything except for the current map because it is already locked
if (pMi == mpCurrentMap) {
  lock = lock_guard<mutex>(pMi -> mMutexMapUpdate)
}
```
 rather than the unused mutex and the ternary however, I discovered this morning that lock_guard has deleted the default constructor so this isn't possible.

Reply:
Oh sorry, I think the line is readable. Rather, I think it's hard to understand that sometimes the reverse lock ordering is safe.